python/ml4ir/applications/ranking/tests/test_feature_fns.py

```diff
@@ -158,3 +158,49 @@ def test_query_length_without_tokenization(self):
         expected_query_len = np.array([7, 11, 3]).reshape(-1, 1, 1)
 
         self.assertTrue(np.isclose(actual_query_len, expected_query_len).all())
+
+    def test_query_type_vector(self):
+        """
+        Test QueryTypeVector feature transformation
+        """
+        feature_info = copy.deepcopy(FEATURE_INFO)
+        feature_info["feature_layer_info"]["args"]["output_mode"] = "one_hot"
+
+        # Define an input string tensor
+        string_tensor = tf.constant(
+            ["", "abc", "123", "!!!", "abc123", "abc@xyz.com", "123.456", "abc2@xyz.com", "\"abc\"", "abc xyz"]
```

Comment:
Notice that the first query is blank and falls into the OOV bucket (0 by default)